main.go

```diff
@@ -25,6 +25,15 @@ import (
 )
 
 func main() {
+	// NOTE(jalextowle): Since `os.Exit` will cause the process to exit, this defer
```

Comment:
TODO comments are fine with a name. But I haven't seen NOTE comments with names. Does this mean that this note has been added by that person ?

To be clear, I have totally no issues with names being added in the code. Just that I think it's a bit odd to have names in a NOTE comment. Usually I would just give a normal comment, and that would suffice.

Reply:
We use this convention within the https://github.com/0xProject/0x-mesh codebase because it documents the people who know the most about particular issues. 

In this case, I think the bug was fairly straightforward, so I'm happy to just leave the comment as an unnamed `NOTE`.